app/controllers/questions_controller.rb

```diff
@@ -0,0 +1,95 @@
+# frozen_string_literal: true
+
+class QuestionsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :authorize_moderator, only: %i[create update destroy]
+  before_action :set_question, only: %i[show edit update destroy]
+  before_action :check_question_bank, only: %i[index show edit update destroy]
+
+  def index
+    @questions = Question.all
+    @questions = Question.paginate(page: params[:page], per_page: 6)
+    @questions = @questions.where(category_id: params[:category_id]) if params[:category_id].present?
+    @questions = @questions.where(difficulty_level: params[:difficulty_level]) if params[:difficulty_level].present?
+    if params[:q].present?
+      @questions = @questions.where("heading LIKE :query OR statement LIKE :query", query: "%#{params[:q]}%")
+    end
+
+    @categories = QuestionCategory.all
+    render :index
+  end
+
+  def check_question_bank
+    return unless Flipper.enabled?(:question_bank)
```

Comment:
Return (allowing the display of the question bank) if the flag is enabled; otherwise, return a status error."

```suggestion
    return if Flipper.enabled?(:question_bank)
```